src/Middleware/CachedResponse.php

```diff
@@ -18,38 +18,138 @@
  * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
  */
 
-declare(strict_types=1);
-
 namespace PrestaShop\Module\DistributionApiClient\Middleware;
 
-class CachedResponse
+use Symfony\Component\HttpClient\Exception\JsonException;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Simple DTO containing the response data to allow serializing it into cache.
+ */
+class CachedResponse implements ResponseInterface
 {
-    /** @var string[][] */
-    private $headers;
+    private int $statusCode;
 
-    /** @var string */
-    private $body;
+    /**
+     * @var string[][]
+     */
+    private array $headers;
+    private string $content;
 
     /**
-     * @param string[][] $headers
-     * @param string $body
+     * @var mixed[]|array|null
      */
-    public function __construct(array $headers, string $body)
+    private ?array $jsonData = null;
+
+    /**
+     * @var array<string, mixed>
+     */
+    private array $info;
+
+    public function __construct(ResponseInterface $response)
+    {
+        $info = $response->getInfo();
+        if (is_array($info)) {
+            $this->info = [
```

Comment:
that's nitpicking but I think this code could be shortened a lot by making an array with all the default "info" values, and then doing an `array_merge` with $info to apply the content of `$response->getInfo()`

there must be a way to do the same with an object with `array_map`

Reply:
(you actually already have the array with default values in the else condition)

Reply:
Yeah this is a bit complex and mostly to please PHPSTan actually 😅

Reply:
damn, alright ! what wouldn't we do for phpstan :D

Reply:
there was probably a better way to solve this, but I admit I was in a rush
and in case the response really return an object at least we're covered